### DISCUSSION TITLE
`$queryRaw` return unexpected type value

### DISCUSSION THREAD

**Discussion by ss49919201** [prisma/prisma#23492] [Q&A] [answered]
### Question

The return type of an aggregate query with `$queryRaw` is unexpected.
I expected number because it is the sum of integers. But object is returned.
Is it possible to return a number?

### How to reproduce (optional)

```typescript
const prisma = new PrismaClient();

type Row = {
  sum: number;
};

const rows = prisma.$queryRaw<
  Row[]
>`SELECT SUM(value) AS sum FROM tbl AS u GROUP BY flag = TRUE`.then((rows) => {
  console.log(typeof rows[0].sum);
});
```

```sh
object
```

### Expected behavior (optional)

```
number
```

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS
- Database: MySQL
- Node: v20.11.1
- Prisma: 5.10.2

schema:

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Tbl {
  id    Int     @id @default(autoincrement())
  value Int
  flag  Boolean

  @@map("tbl")
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @ss49919201 👋 

Thank you for raising this question.

Can you try this query?

```js
const rows = prisma.$queryRaw<Row[]>`SELECT CAST(SUM(value) AS INTEGER) AS sum FROM tbl AS u GROUP BY flag = TRUE`.then((rows) => {
  console.log(typeof rows[0].sum);
});
```
With the query above, I am casting the sum to an integer in your SQL query. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
